### PR TITLE
Feature input validator

### PR DESCRIPTION
**Pull request by grace000**
The Input Validator class checks that incoming moves are appropriate characters (integers), and works with the board to check that spaces are available for incoming moves.

### CODE REVIEW

File: lib/input_validator.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require_relative "board"
```
Comment (BWoodfork) [inline lib/input_validator.rb]: Why's this require here?

File: lib/board.rb [hunk 2]
```diff
@@ -1,5 +1,8 @@
 
 class Board
+  
+    attr_accessor :moves
+  
     def display
         "    1    |  2   |   3   #{horizontal_bar}    4    |  5   |   6   #{horizontal_bar}    7    |  8   |   9  \n"
     end
```
Comment (BWoodfork) [inline lib/board.rb]: We discussed before that Board should not be responsible for displaying itself. That responsibility should live in another class.

File: lib/input_validator.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            return false
+        end
+        true
+    end
+
+    def get_invalid_moves(board)
+        board.moves.each do |i|
+            @invalid_moves << i 
+        end
+    end
```
Comment (BWoodfork) [inline lib/input_validator.rb]: Why would you want a list of invalid moves?

File: lib/input_validator.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+require_relative "board"
+
+class InputValidator
+    def initialize
+        @invalid_moves = []
+    end
+
+    def valid_character?(move)
+        regex = /\D/
```
Comment (BWoodfork) [inline lib/input_validator.rb]: What's a more descriptive name for this?

File: spec/input_validator_spec.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    end
+
+    describe "#valid_move?" do
+        it "should return false if requested move is already in the board's moves array" do
+            board = Board.new
+
+            board.move(2)
+            @input_v.get_invalid_moves(board)
+            expect(@input_v.valid_move?(2)).to eq(false)
+        end
```
Comment (BWoodfork) [inline spec/input_validator_spec.rb]: Where's the opposite test case for this? I.E. when the function does not return false.

File: lib/input_validator.rb (hunk 1 above)
Comment (grace000) [reply lib/input_validator.rb]: I thought I needed that for the get_valid_moves method, but the tests are passing without it. I'll delete that line.

File: lib/board.rb (hunk 2 above)
Comment (grace000) [reply lib/board.rb]: I'll make a new class for rendering the board.

File: lib/input_validator.rb (hunk 3 above)
Comment (grace000) [reply lib/input_validator.rb]: This was my attempt to have the class (on its own) check moves that could not be requested. Ultimately this class will determine whether or not a move can be taken.

File: lib/input_validator.rb (hunk 4 above)
Comment (grace000) [reply lib/input_validator.rb]: Maybe valid_coordinate, valid_position, valid_selection?

File: lib/input_validator.rb (hunk 3 above)
Comment (grace000) [reply lib/input_validator.rb]: Now that I'm looking at your last comment on this PR, I see why this way validating might cause issues.

**Pull request merged**